plugins/woocommerce/src/Internal/DataStores/Orders/FeatureController.php

```diff
@@ -0,0 +1,324 @@
+			$generate_item_return = __( 'Custom orders table is being regenerated', 'woocommerce' );
+			$generate_item_button = __( 'Regenerate', 'woocommerce' );
+		} else {
+			$generate_item_name   = __( 'Create and fill custom orders table', 'woocommerce' );
+			$generate_item_desc   = __( 'This tool will create the custom orders table and fill it with existing orders data from the posts table. This process may take a while.', 'woocommerce' );
+			$generate_item_return = __( 'Custom orders table is being filled', 'woocommerce' );
+			$generate_item_button = __( 'Create', 'woocommerce' );
+		}
+
+		$entry = array(
+			'name'             => $generate_item_name,
+			'desc'             => $generate_item_desc,
+			'requires_refresh' => true,
+			'callback'         => function() use ( $generate_item_return ) {
+				$this->initiate_regeneration_from_tools_page();
+				return $generate_item_return;
+			},
+		);
+
+		if ( $generation_is_in_progress ) {
+			$entry['button'] = sprintf(
+			/* translators: %d: How many orders have been processed so far. */
+				__( 'Filling in progress (%d)', 'woocommerce' ),
+				$this->data_synchronizer->get_regeneration_processed_orders_count()
+			);
+			$entry['disabled'] = true;
+		} else {
+			$entry['button'] = $generate_item_button;
+		}
+
+		$tools_array['regenerate_custom_orders_table'] = $entry;
+
+		if ( $orders_table_exists ) {
+
+			// Delete the table.
+
+			$tools_array['delete_custom_orders_table'] = array(
+				'name'             => __( 'Delete the custom orders table', 'woocommerce' ),
+				'desc'             => sprintf(
+					'<strong class="red">%1$s</strong> %2$s',
+					__( 'Note:', 'woocommerce' ),
+					__( 'This will delete the custom orders table. You can create it again with the "Create and fill custom orders table" tool.', 'woocommerce' )
+				),
+				'button'           => __( 'Delete', 'woocommerce' ),
+				'requires_refresh' => true,
+				'callback'         => function () {
+					$this->delete_custom_orders_table();
+					return __( 'Custom orders table has been deleted.', 'woocommerce' );
+				},
+			);
+		}
+
+		return $tools_array;
+	}
+
+	/**
+	 * Initiate the custom orders table (re)generation in response to the user pressing the tool button.
+	 *
+	 * @throws \Exception Can't initiate regeneration.
+	 */
+	private function initiate_regeneration_from_tools_page() {
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput
+		if ( ! isset( $_REQUEST['_wpnonce'] ) || false === wp_verify_nonce( $_REQUEST['_wpnonce'], 'debug_action' ) ) {
+			throw new \Exception( 'Invalid nonce' );
+		}
+
+		$this->check_can_do_table_regeneration();
+		$this->data_synchronizer->initiate_regeneration();
+	}
+
+	/**
+	 * Can the custom orders table regeneration be started?
+	 *
+	 * @throws \Exception The table regeneration can't be started.
+	 */
+	private function check_can_do_table_regeneration() {
+		if ( ! $this->$this->is_feature_visible() ) {
```

Comment:
```suggestion
		if ( ! $this->is_feature_visible() ) {
```
seems like an extra `$this` sneaked in